lib/assembly/build.ex

```diff
@@ -0,0 +1,35 @@
+defmodule Assembly.Build do
+  use GenServer
+
+  alias Assembly.{Builder, Components}
+  alias Bottle.Assembly.V1.Build
+
+  def start_link(%Build{} = build) do
+    GenServer.start_link(__MODULE__, build)
+  end
+
+  @impl true
+  def init(%Build{} = build) do
+    {:ok, build}
+  end
+
+  @impl true
+  def handle_cast(:determine_status, %Build{component: components, status: current_status} = build) do
+    readyable? =
+      Enum.all?(components, fn %{id: component_id} ->
+        {:ok, quantity} = Components.quantity_available(component_id)
+        not is_nil(quantity) and quantity > 0
```

Comment:
This is almost certainly not the correct logic, it's just here to illustrate this is where we will do the computation on demand vs availability.